src/main.rs

```diff
@@ -53,17 +53,26 @@ fn main() {
         }))
         .header("Authorization", format!("Bearer {}", config.api_key))
         .send()
-        .unwrap()
-        .error_for_status()
-        .unwrap_or_else(|_| {
-            spinner.stop_and_persist(
-                "✖".red().to_string().as_str(),
-                "Failed to get a response. Have you set the OPENAI_API_KEY variable?"
-                    .red()
-                    .to_string(),
-            );
-            std::process::exit(1);
-        });
+        .unwrap();
+
+    let status_code = response.status();
+    if status_code.is_client_error() {
```

Comment:
I wasn't able to get the response body after using `error_for_status`. It looks to me like the `Error` type does not expose the response body. Since `error_for_status` is not doing much more than turning the response into ok / error by using `if status.is_client_error() || status.is_server_error() {`, I just use those methods here directly.